apis/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,174 @@
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
+
+	// List of sources whose traffic this AdminNetworkPolicyRule applies to.
+	// If any adminNetworkPolicyPeer matches the source of incoming
+	// traffic then the specified action is applied.
+	// This field must be defined and contain at least one item.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	From []AdminNetworkPolicyPeer `json:"from"`
+}
+
+// AdminNetworkPolicyEgressRule describes an action to take on a particular
+// set of traffic originating from pods selected by a AdminNetworkPolicy's
+// Subject field. The traffic must match both ports and to.
+type AdminNetworkPolicyEgressRule struct {
+	// Name is an identifier for this rule, that may be no more than 100 characters
+	// in length. This field should be used by the implementation to help
+	// improve observability, readability and error-reporting for any applied
+	// AdminNetworkPolicies.
+	// +optional
+	// +kubebuilder:validation:MaxLength=100
+	Name string `json:"name,omitempty"`
+
+	// Action specifies the affect this rule will have on matching traffic,
+	// currently the following actions are supported:
+	// Allow: allows the selected traffic (even if it would otherwise have been denied by NetworkPolicy)
+	// Deny: denies the selected traffic (even if it would otherwise have been denied by NetworkPolicy)
+	// Pass: instructs the selected traffic to skip any remaining ANP rules, and
+	// then pass execution to any NetworkPolicies that select the pod.
+	// If the pod is not selected by any NetworkPolicies then execution
+	// is passed to any BaselineAdminNetworkPolicies that select the pod.
+	// This field is mandatory.
+	Action AdminNetworkPolicyRuleAction `json:"action"`
+
+	// Ports allows for matching traffic based on port and protocols.
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
+
+	// List of destinations whose traffic this adminNetworkPolicyRule applies to.
+	// If any adminNetworkPolicyPeer matches the destination of outgoing
+	// traffic then the specified action is applied.
+	// This field must be defined and contain at least one item.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	To []AdminNetworkPolicyPeer `json:"to"`
+}
+
+// AdminNetworkPolicyRuleAction string describes the AdminNetworkPolicy action type.
+// +enum
+type AdminNetworkPolicyRuleAction string
+
+const (
+	// AdminNetworkPolicyRuleActionPass enables admins to provide exceptions to
+	// AdminNetworkPolicies by passing rule execution directly to any matching
+	// K8s networkPolicies.
+	AdminNetworkPolicyRuleActionPass AdminNetworkPolicyRuleAction = "Pass"
```

Comment:
```suggestion
	// AdminNetworkPolicyRuleActionPass indicates that matching traffic will
	// bypass further AdminNetworkPolicy processing (ignoring rules with lower
	// precedence) and be allowed or denied based on NetworkPolicy and
	// BaselineAdminNetworkPolicy rules.
	AdminNetworkPolicyRuleActionPass AdminNetworkPolicyRuleAction = "Pass"
```

also, probably this section should be in the order Allow, Deny, Pass, like it is in the descriptions of Action above?